rest_framework/fields.py

```diff
@@ -71,7 +71,11 @@ def get_attribute(instance, attrs):
         except ObjectDoesNotExist:
             return None
         if is_simple_callable(instance):
-            instance = instance()
+            try:
+                instance = instance()
+            except (AttributeError, KeyError) as exc:
+                raise CallableAttributeExceptionWrapper(exc)
```

Comment:
Rather than wrap these in a special exception type that's handled explicitly it might be simpler to just wrap them in, say, `ValueError`, with a descriptive message along the lines "Invalid value for 'default' argument. An exception was raised. Original message was: ...".

Let's then just keep that super simple and not bother catching it elsewhere - at least we're giving better visibility to the developer then without adding complexity to the `Field.get_attribute()` method.